org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -50,6 +50,7 @@
 
     private int unknown_state_count;
     private boolean step_handshake;
+    private boolean returned_finished;
```

Comment:
@edewata Can you think of anything more elegant for this? Or is adding a boolean flag and tracking it fine? Maybe I should split it into a `updateHandshakePre()` and a `updateHandshakeState()`? -- `Pre` only getting called once and `State` getting called only when actually handshaking? Hmm...

Reply:
I don't really have an opinion on this. If it works then let's keep it :)
We can always refactor later if we find a better way.

I've ACKed this PR, feel free to merge, but I'd prefer @jmagne to take a look at this as well, either before or after merging.